src/Npgsql/PostgresMinimalDatabaseInfo.cs

```diff
@@ -15,16 +15,29 @@ class PostgresMinimalDatabaseInfoFactory : INpgsqlDatabaseInfoFactory
         => Task.FromResult(
             conn.Settings.ServerCompatibilityMode == ServerCompatibilityMode.NoTypeLoading
                 ? (NpgsqlDatabaseInfo)new PostgresMinimalDatabaseInfo(conn)
-                : null
-        );
+                : null);
 }
 
 class PostgresMinimalDatabaseInfo : PostgresDatabaseInfo
 {
-    static readonly PostgresBaseType[] Types = typeof(NpgsqlDbType).GetFields()
+    static readonly PostgresType[] Types = typeof(NpgsqlDbType).GetFields()
         .Select(f => f.GetCustomAttribute<BuiltInPostgresType>())
         .OfType<BuiltInPostgresType>()
-        .Select(a => new PostgresBaseType("pg_catalog", a.Name, a.OID))
+        .SelectMany(attr =>
+        {
+            var baseType = new PostgresBaseType("pg_catalog", attr.Name, attr.BaseOID);
+            var arrayType = new PostgresArrayType("pg_catalog", "_" + attr.Name, attr.ArrayOID, baseType);
+
+            if (attr.RangeName is null)
+            {
+                return new PostgresType[] { baseType, arrayType };
+            }
+
+            var rangeType = new PostgresRangeType("pg_catalog", attr.RangeName, attr.RangeOID, baseType);
+            var multirangeType = new PostgresMultirangeType("pg_catalog", attr.MultirangeName!, attr.MultirangeOID, rangeType);
```

Comment:
Shouldn't we return multirange type only if PG's version >= 14?

Reply:
Done